### PR TITLE
[hermes] Add more rest api methods

### PR DESCRIPTION
**Pull request by ali-bahjati**
Add many of the price service apis. Per David suggestion, we do validation in parsing instead of doing it later. I didn't find any suitable library to deserialize our hex format so I created a macro to implement it because we use it in a couple of places. I tried making a generic HexInput but couldn't make it working (and I need other crates like generic_array for it which makes the code more complex)

### CODE REVIEW

File: hermes/src/network/rpc/rest.rs [hunk 1]
```diff
@@ -107,6 +102,91 @@ pub async fn latest_price_feeds(
...
+impl_deserialize_for_hex_string_wrapper!(GetVaaCcipInput, 40);
+
+#[derive(Debug, serde::Deserialize)]
+pub struct GetVaaCcipQueryParams {
+    data: GetVaaCcipInput,
+}
+
+#[derive(Debug, serde::Serialize)]
+pub struct GetVaaCcipResponse {
+    data: String,
```
Comment (Reisen) [inline hermes/src/network/rpc/rest.rs:162]: TODO: Add type.

File: hermes/src/network/rpc/rest.rs [hunk 2]
```diff
@@ -24,47 +15,56 @@ use {
...
+    },
+    pyth_sdk::{
+        PriceFeed,
+        PriceIdentifier,
+    },
 };
 
+#[derive(Debug, Clone, Deref, DerefMut)]
+pub struct PriceIdInput([u8; 32]);
+impl_deserialize_for_hex_string_wrapper!(PriceIdInput, 32);
```
Comment (Reisen) [inline hermes/src/network/rpc/rest.rs:34]: May be better to replace with `struct Hex` with const generics.

File: hermes/src/network/rpc/rest.rs [hunk 3]
```diff
@@ -107,6 +102,91 @@ pub async fn latest_price_feeds(
...
+
+    let price_feeds_with_update_data = state
+        .store
+        .get_price_feeds_with_update_data(vec![price_id], RequestTime::FirstAfter(publish_time))
+        .map_err(|_| RestError::UpdateDataNotFound)?;
+
+    let vaa = price_feeds_with_update_data
+        .update_data
+        .batch_vaa
+        .get(0)
```
Comment (Reisen) [inline hermes/src/network/rpc/rest.rs:180]: Add comment explaining why `get(0)` is safe, and the semantics behind the batch structure.

**Review by Reisen — CHANGES_REQUESTED** [review]

**Review by Reisen — APPROVED** [review]

File: hermes/src/network/rpc/rest.rs (hunk 2 above)
Comment (ali-bahjati) [reply hermes/src/network/rpc/rest.rs:34]: I will address this in a later PR. Added a todo for it.

**Pull request merged**